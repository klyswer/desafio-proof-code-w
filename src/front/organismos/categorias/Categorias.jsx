import React from 'react';
import PropTypes from 'prop-types';
import ItemCategoria from '../../atomos/itemCategoria/ItemCategoria';
import './style.css';

const Categorias = ({listaCategorias=[]}) => {
  return (
    listaCategorias !== [] ? (
      <div className="container_categorias">
        {
          listaCategorias.map((elm) => (
            <div key={elm.id} className="">
              <ItemCategoria svg={elm.icon} title={elm.title} active={elm.active} />
            </div>
            )
          )
        }
      </div>
    ) : (
      <h5>Cargando Categorias</h5>
    )
  );
};


Categorias.propTypes = {
  listaCategorias:PropTypes.arrayOf(PropTypes.object).isRequired,
};


export default Categorias;