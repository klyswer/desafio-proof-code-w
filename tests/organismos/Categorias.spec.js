import React from 'react';
/* eslint-disable no-undef */
import { mount } from 'enzyme';
import Categorias from '../../src/front/organismos/categorias/Categorias';


describe('<Categorias />', () => {

  const productos = [
      {id:1,title:'Juguetería',icon:'jugueteria.svg',active:false},
      {id:2,title:'Mundo Bebé',icon:'bebe.svg',active:false}
  ];

  const wrapper = mount(<Categorias listaCategorias={productos} />);

  test('Primer Snapshot: <Categorias />', () => { 
    expect(wrapper).toMatchSnapshot();
  });
  
});
