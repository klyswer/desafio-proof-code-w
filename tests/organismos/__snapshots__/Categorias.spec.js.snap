// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categorias /> Primer Snapshot: <Categorias /> 1`] = `
<div
  className="container_categorias"
>
  <div
    key="1Juguetería"
  >
    <div
      className="item_default_categoria false"
    >
      <div
        className="contenedor_elm_categoria"
      >
        <div
          className="icon_categoria"
        >
          <img
            alt="Juguetería"
            src="/images/jugueteria.svg"
          />
        </div>
        <div
          className="texto_categoria"
        >
          Juguetería
        </div>
      </div>
    </div>
  </div>
  <div
    key="2Mundo Bebé"
  >
    <div
      className="item_default_categoria false"
    >
      <div
        className="contenedor_elm_categoria"
      >
        <div
          className="icon_categoria"
        >
          <img
            alt="Mundo Bebé"
            src="/images/bebe.svg"
          />
        </div>
        <div
          className="texto_categoria"
        >
          Mundo Bebé
        </div>
      </div>
    </div>
  </div>
</div>
`;
